packages/devtools_app/lib/src/flutter/common_widgets.dart

```diff
@@ -0,0 +1,75 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_widgets/flutter_widgets.dart';
+
+/// Convenience [Divider] with [Padding] to fit in better with forms.
+class PaddedDivider extends StatelessWidget {
+  const PaddedDivider({
+    Key key,
+    this.padding = const EdgeInsets.only(bottom: 10.0),
+  }) : super(key: key);
+
+  /// The padding to place around the divider.
+  final EdgeInsets padding;
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: padding,
+      child: const Divider(thickness: 1.0),
+    );
+  }
+}
+
+/// A [TaggedText] with builtin DevTools-specific text styling.
+class DevToolsTaggedText extends StatelessWidget {
```

Comment:
I would like us to avoid prefixing class names with the word DevTools when possible.
Is there a name that specifies what this does and when it should be used?
For example, is this DefaultTaggedText that gives you styled text matching the defaultTextStyle from devtools along with a default set of tags convenient for porting the app?

Lets also document this class with what cases we expect people porting should use it versus define widgets directly instead of using HTML. I imagine the TaggedText class is ok when performance isn't a concern but has negative performance and error checking functionality relative to just using Flutter widgets directly.

Reply:
Name change SGTM.

Documentation: This widget is a wrapper around Flutter's RichText widget.  That widget takes TextSpans, not other widgets.

It's an alternative to that for richly-formatted text.  The performance is roughly the same, and it will throw assertion errors in any cases where the text isn't parsed properly.

The benefits to using it are that the xml styling is much easier to read than creating multiple TextSpans in a RichText.

It doesn't do widgets, and it doesn't do HTML, just rich text.